src/App.tsx

```diff
@@ -1,19 +1,107 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { getTodos } from './api/todos';
+import { NewTodoField } from './components/NewTodoField/NewTodoField';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { FilterType } from './types/Filtertype';
+import { ErrorType } from './types/ErrorType';
+import { Error } from './components/Error/Error';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+
+  const [filterType, setFilterType] = useState(FilterType.all);
+  const [activeTodoLength, setActiveTodosLength] = useState(0);
```

Comment:
why do we need this state? just create variable, wrap with usememo, and calculate length with `filter` method